plugins/influxdb/influxdb.go

```diff
@@ -0,0 +1,142 @@
+package influxdb
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/Sirupsen/logrus"
+	"github.com/stripe/veneur/plugins"
+	"github.com/stripe/veneur/samplers"
+)
+
+var _ plugins.Plugin = &InfluxDBPlugin{}
+
+// InfluxDBPlugin is a plugin for emitting metrics to InfluxDB.
+type InfluxDBPlugin struct {
+	Logger     *logrus.Logger
+	InfluxURL  string
+	HTTPClient *http.Client
+	Statsd     *statsd.Client
+}
+
+// NewInfluxDBPlugin creates a new Influx Plugin.
+func NewInfluxDBPlugin(logger *logrus.Logger, addr string, consistency string, db string, client *http.Client, stats *statsd.Client) *InfluxDBPlugin {
+	plugin := &InfluxDBPlugin{
+		Logger:     logger,
+		HTTPClient: client,
+		Statsd:     stats,
+	}
+
+	inurl, err := url.Parse(addr)
+	if err != nil {
+		logger.Fatalf("Error parsing URL for InfluxDB: %q", err)
+	}
+
+	// Construct a path we will be using later.
+	inurl.Path = "/write"
+	q := inurl.Query()
+	q.Set("db", db)
+	q.Set("precision", "s")
+	inurl.RawQuery = q.Encode()
+	plugin.InfluxURL = inurl.String()
+
+	return plugin
+}
+
+// Flush sends a slice of metrics to InfluxDB
+func (p *InfluxDBPlugin) Flush(metrics []samplers.DDMetric, hostname string) error {
+	p.Statsd.Gauge("flush.post_metrics_total", float64(len(metrics)), nil, 1.0)
+	// Check to see if we have anything to do
+	if len(metrics) == 0 {
+		p.Logger.Info("Nothing to flush, skipping.")
+		return nil
+	}
+
+	buff := bytes.Buffer{}
```

Comment:
We're writing to it for each member of `metrics`, so we can't use `strings.Reader` here, right? TIL about `WriteString` though, so I switched to it to avoid the ugliness of the inline `[]byte`.